### ISSUE TITLE
Mysterious constant in addition to all trees

### ISSUE THREAD

**Issue by liujxing** [dmlc/xgboost#3341]
I trained a xgboost model and manually checked all trees. Sometimes the final forecast has a constant (in my case it is 0.5, but sometimes it is 0) added to all predictions, and such constant is not specified by any of the trees. Where does this constant come from? The margin I specified when training the model is always 0.

**Comment by yanboliang** [comment]
There is a parameter ```base_score``` which is the global bias of the model, its [default value is 0.5](https://github.com/dmlc/xgboost/blob/ccf80703efefc7c9ff988c454c05e01beacba26d/src/learner.cc#L59). For highly imbalanced data you can initialize it to a more meaningful base score. Theoretically, as long as you choose the right learning rate and give it enough steps to train, it would not affect the final result.
